### ISSUE TITLE
Memory Clear Unserialized values.

### ISSUE THREAD

**Issue by Brandantl** [USCiLab/cereal#333]
Is there such a way to memset values on load only using 
`void serialize(Archive & ar) { ar(...) }`

For example, let's say I'm using saving this structure

```
struct asdf {
int a;
int b; 
int c;
}
```
Now c is useless information and not worth saving, but it's required to be predictable for networking synchronization.

Now let's say it has a serialize function which covers A and B.
`void serialize(Archive & ar) { ar(a, b) }`

is c undefined, therefore, unpredictable what its value will be upon reloading this struct? So if you load this on 2 machines can C be 5 and 8 when loading the same file? 

If so how can I detect that we're loading a serialized struct and set C to 0?

Basically what I'm asking is how can I detect if Archive is XML read or XML write so I can initialize the values we're not saving to save space.

**Comment by nabijaczleweli** [comment]
Initialise C to 0 beforehand?

**Comment by erichkeane** [comment]
If you need to set C, do it in the serialization functions.
If you need to tell the diference between serializing and deserializing, use the 'load' and 'save' functions rather than 'serialize'.  See the right-most code on the first code example here: http://uscilab.github.io/cereal/serialization_functions.html

**Comment by Brandantl** [comment]
Thats exactly what I was looking for, I didn't see that. Thanks!
